### ISSUE TITLE
Mysql DB Connection expires

### ISSUE THREAD

**Issue by Otlichno** [Segs/Segs#579]
### Expected behavior
Connection to the database to be kept alive
### Actual behavior/symptoms
MapServer 1 now listening on 149.56.101.186 : 7003
Debug   : User Otlichno trying to login from x.x.x.x
Debug   : SQL_ERROR: QSqlError("2006", "QMYSQL3: Unable to reset statement", "MySQL server has gone away")
Debug   : AuthHandler Reaping stale link 140358524620720
Debug   : User Otlichno trying to login from x.x.x.x
Debug   : SQL_ERROR: QSqlError("2013", "QMYSQL3: Unable to reset statement", "Lost connection to MySQL server during query")
Debug   : AuthHandler Reaping stale link 140358524619728
Debug   : User Otlichno trying to login from x.x.x.x
Debug   : SQL_ERROR: QSqlError("2013", "QMYSQL3: Unable to reset statement", "Lost connection to MySQL server during query")
Debug   : AuthHandler Reaping stale link 140358524619728
### SEGS version
commit 16c86cfdcc069353b4707b35b4b0efe6eb1b63c3
Merge: 15317d4 595227b
Author: Lloyd Dilley <ldilley@users.noreply.github.com>
Date:   Fri Sep 14 22:46:26 2018 -0400

    Merge pull request #573 from cyanpill/develop

     Starting character infromation read from settings.cfg

### Operating system
CentOS 7

**Comment by broxen** [comment]
It appears this happens when the DB server has not had any activity for 8+ hrs.

As long as the server is online, we should ping the DB server occasionally to maintain connection.

Citing these references:
https://forum.qt.io/topic/59960/mysql-lost-connection-solved
https://dev.mysql.com/doc/refman/8.0/en/gone-away.html

**Comment by dracc** [comment]
Another "solution" would be to setup the MySQL server to not kill the connection as early - but I'd consider that more of a hack than an actual fix.

**Comment by miles200** [comment]
Late to the party... could we run a simple/lightweight query every 7 hours or so?

**Comment by broxen** [comment]
That seems like a great idea

**Comment by nemerle** [comment]
Yes, yes we should simply run a timer that runs a very simple query ( `select true` ) every 30 seconds.
This will work for all server-based db engines as well ( pgsql/mysql )
